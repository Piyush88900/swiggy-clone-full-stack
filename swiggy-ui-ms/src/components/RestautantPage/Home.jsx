import React from 'react';
import Navigator from './Navigator';
import HomeDummy from './HomeDummy';
import AllRestaurants from './AllRestaurants';

function Home() {
    return (
        <div>
            <Navigator />
            <HomeDummy />
            <AllRestaurants />
        </div>
    );
}

export default Home;
