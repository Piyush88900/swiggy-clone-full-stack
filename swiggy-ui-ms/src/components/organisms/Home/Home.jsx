/**
 *
 * Home
 *
 */
import React from 'react';
import { useQuery } from '@apollo/client';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { getCurrency } from '../../../graphQL/Queries/Currency';

const Home = () => {
  const { data, loading } = useQuery(getCurrency);

  if (loading) {
    return '...loading content';
  }

  return (
    <TableContainer component={Paper}>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Rate</TableCell>
            <TableCell align="right">Currency</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.rates.map(({ rate, currency }) => (
            <TableRow key={`rate_${currency}`}>
              <TableCell>{Math.round(rate * 100) / 100}</TableCell>
              <TableCell align="right">{currency}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default Home;
