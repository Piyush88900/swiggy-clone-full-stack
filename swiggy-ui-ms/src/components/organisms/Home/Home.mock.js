const mockResponse = {
  rates: [
    { currency: 'AED', rate: 0.049459055550450607793, __typename: 'ExchangeRate' },
    { currency: 'AFN', rate: 1.036849272067195181312882, __typename: 'ExchangeRate' },
  ],
};

export default mockResponse;
