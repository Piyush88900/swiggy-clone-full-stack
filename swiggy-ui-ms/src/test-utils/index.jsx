import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { ThemeProvider } from '@material-ui/core/styles';
import { LabelProvider } from '../stores/labels';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>
        <LabelProvider>{children}</LabelProvider>
      </ThemeProvider>
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
