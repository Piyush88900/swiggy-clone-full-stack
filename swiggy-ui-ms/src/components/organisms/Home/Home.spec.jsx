import React from 'react';
import { act } from '@testing-library/react';
import { render } from '../../../test-utils';
import { getCurrency } from '../../../graphQL/Queries/Currency';
import mockResponse from './Home.mock';

import Home from '.';

describe('Home', () => {
  it('should render loading state correctly', () => {
    const { asFragment } = render(<Home />, {
      wrapperProps: {
        mocks: [
          {
            request: {
              query: getCurrency,
            },
            result: {
              data: mockResponse,
            },
          },
        ],
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly once data is loaded', async () => {
    const { asFragment } = render(<Home />, {
      wrapperProps: {
        mocks: [
          {
            request: {
              query: getCurrency,
            },
            result: {
              data: mockResponse,
            },
          },
        ],
      },
    });

    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
