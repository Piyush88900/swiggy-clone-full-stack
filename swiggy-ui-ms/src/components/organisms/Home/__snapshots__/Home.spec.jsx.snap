// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly once data is loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            Rate
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            Currency
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            0.05
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            AED
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1.04
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            AFN
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Home should render loading state correctly 1`] = `
<DocumentFragment>
  ...loading content
</DocumentFragment>
`;
